src/gui/instrument/EnvelopeGraph.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * EnvelopeGraph.cpp - Displays envelope graphs
+ *
+ * Copyright (c) 2004-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2024-     Michael Gregorius
+ * 
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "EnvelopeGraph.h"
```

Comment:
I have applied this in another branch which is a follow-up PR to this one. The branch completely rewrites the rendering of the envelope graph so that it can render at any widget size. Done there to prevent unnecessary merge conflicts and confusion.